### ISSUE TITLE
Integration of White Papers into mainline documentation

### ISSUE THREAD

**Issue by aallan** [raspberrypi/documentation#2973]
We need to properly integrate the White Papers into the mainline documentation.

The first thing here is to do another pass through the existing white papers to see if there is anything that should be surfaced more generally, and therefore needs to get poured into the documentation rather than exist as a separate document.

Next however is adding an info block to the online documentation to point users at white papers that extend the documentation. All the white papers should (in theory) be extensions — deep dives into certain topics — extending the existing documentation. We should therefore insert info blocks at the correct places in the documentation pointing users to that extended docuemntation.

I'm sort of envisioning a new type of [admonitions](https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#admonitions)-style block for emphasis called `[WHITEPAPER]` which might look something like this,

<img width="1384" alt="Screenshot 2023-06-02 at 04 43 39" src="https://github.com/raspberrypi/documentation/assets/53625/854c6980-64c9-4441-8395-a1ee04880e39">

although if we want to do an initial fast pass where we can start by using a standard `[INFO]` block to get them all integrated more quickly.

My understanding is that we can't add a new admonition-style block, so we'd probably have to add a macro and handle styling in pre-processing. I'd suggest it would look like this,

````
[whitepaper, title="White Paper Title", subtitle="A subtitle for the white paper", link=https://url.here]
----
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus elementum purus a eros laoreet, a vestibulum risus interdum. Suspendisse bibendum sagittis felis et pulvinar. Fusce purus ante, vestibulum in lacinia a, placerat quis libero. Nunc nibh mauris, finibus vel fermentum vulputate, convallis id neque. In in rhoncus urna. Aliquam sed sapien porta, venenatis massa vitae, aliquam mauris. Nunc ut ante pharetra, faucibus ante convallis, venenatis ligula. 

Nullam volutpat, sem quis commodo condimentum, libero ex aliquam urna, in aliquet justo felis mattis libero. Pellentesque vestibulum enim vel augue blandit tempus. Ut a elit non turpis viverra aliquam sed a felis. Etiam ut porta mi. Sed feugiat arcu leo, vulputate placerat ante ullamcorper at. In id metus sit amet massa molestie efficitur. Morbi rutrum scelerisque metus id dignissim.
----

where the title, subtitle and link are automagically inserted into the text in the block.

**Comment by aallan** [comment]
Ping @nelliemckesson and @lurch and @JamesH65.

First thing to do would be to get a list of all published White Papers in PIP and see where they extend the existing online documentation (@JamesH65) so we can insert an appropriate `[INFO]` block directing users to the extended docs.

Next thing to do would be @nelliemckesson and @lurch taking a look at how to implement the styling of a new block. Sounds like a @nelliemckesson thing in the first instance there.

**Comment by aallan** [comment]
Maybe something like this would be cleaner, with the link being the title?

<img width="1372" alt="Screenshot 2023-06-02 at 09 26 52" src="https://github.com/raspberrypi/documentation/assets/53625/96e53fcc-ca9d-4c7b-803f-67af9c42dee2">

**Comment by JamesH65** [comment]
The current list of WPs is here, https://pip.raspberrypi.com/categories/685-whitepapers-app-notes

**Comment by JamesH65** [comment]
> Maybe something like this would be cleaner, with the link being the title?
> 
> ![Screenshot 2023-06-02 at 09 26 52](https://user-images.githubusercontent.com/53625/242831694-2caadec9-3326-4cf2-894e-6462d604d864.jpg)

It would be cool if clicking on the image of the front page also follows the link.

**Comment by JamesH65** [comment]
It may or may not be useful but all WPs have metadata in them with title and subtitle, so they could be extracted from the PDF rather than being retyped in the docs. 
![image](https://github.com/raspberrypi/documentation/assets/3844893/71b7e3e3-decf-4238-8a16-25d8b804a3a8)

**Comment by nelliemckesson** [comment]
@aallan In the interest of good HTML semantics, I'd propose we use a sidebar with a custom role for these, like this:

```
[.whitepaper, title="White Paper Title", subtitle="A subtitle for the white paper", link=https://url.here]
****
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus elementum purus a eros laoreet, a vestibulum risus interdum. Suspendisse bibendum sagittis felis et pulvinar. Fusce purus ante, vestibulum in lacinia a, placerat quis libero. Nunc nibh mauris, finibus vel fermentum vulputate, convallis id neque. In in rhoncus urna. Aliquam sed sapien porta, venenatis massa vitae, aliquam mauris. Nunc ut ante pharetra, faucibus ante convallis, venenatis ligula. 

Nullam volutpat, sem quis commodo condimentum, libero ex aliquam urna, in aliquet justo felis mattis libero. Pellentesque vestibulum enim vel augue blandit tempus. Ut a elit non turpis viverra aliquam sed a felis. Etiam ut porta mi. Sed feugiat arcu leo, vulputate placerat ante ullamcorper at. In id metus sit amet massa molestie efficitur. Morbi rutrum scelerisque metus id dignissim.
****
```

But whichever block type we choose, adding a custom role would be really helpful.

Jekyll/asciidoctor have built-in handling for custom templates for specific elements or blocks. I threw together [an initial draft here](https://github.com/raspberrypi/documentation/pull/2975).

The sidebar markup above plus the changes in the PR linked above generate output like this (I'm auto-generating the cover image here -- all the CSS could use a little fine-tuning but I wanted to show you a rough idea of what we could do):
 
<img width="1419" alt="Screen Shot 2023-06-06 at 11 50 13 AM" src="https://github.com/raspberrypi/documentation/assets/812739/d55ee968-711f-4cc7-8ad3-ee66ecc1cbc2">

**Comment by aallan** [comment]
> I'd propose we use a sidebar with a custom role for these, like this…

This works for me. If I merge the draft PR this is what we get? Although I have a feeling the CSS will inevitably need to be tweaked, I'm pretty happy with the above. Tempted to merge and get the published White Papers properly integrated into the docs this week. Are you happy for me to go ahead and merge @nelliemckesson ?

**Comment by aallan** [comment]
> It may or may not be useful but all WPs have metadata in them with title and subtitle, so they could be extracted from the PDF rather than being retyped in the docs.

Huh. Nice idea, but a bit worried by adding too much magic in at this stage.

**Comment by aallan** [comment]
> It would be cool if clicking on the image of the front page also follows the link.

Is this doable @nelliemckesson ?

**Comment by aallan** [comment]
> It would be cool if clicking on the image of the front page also follows the link.

Is this doable @nelliemckesson ?

**Comment by nelliemckesson** [comment]
@aallan yeah should be doable. I’ll add that today and then we can merge!

**Comment by aallan** [comment]
…and merged #2975

**Comment by aallan** [comment]
I'll have a poke around and get a test case WP inline in the documentation tomorrow. Thanks @nelliemckesson.

**Comment by aallan** [comment]
Hey @nelliemckesson. Can you add a bit more margin below white paper box, the following text seems to be a bit crushed up to it in my initial experimentations with things?

<img width="931" alt="Screenshot 2023-06-13 at 14 14 23" src="https://github.com/raspberrypi/documentation/assets/53625/f241de68-e7a3-4fcd-b22f-30811b335cc1">

**Comment by aallan** [comment]
Maybe just throw that change directly onto develop?

**Comment by aallan** [comment]
…and fixed in #2981

**Comment by aallan** [comment]
Looks like there is a problem with styling on production.

The white paper info boxes work in the local browser version, but don’t render correctly on the production server.

e.g. see https://raw.githubusercontent.com/raspberrypi/documentation/develop/documentation/asciidoc/computers/raspberry-pi/raspberry-pi-industrial.adoc which has a white paper block at the top.

This renders like this locally,

<img width="893" alt="Screenshot 2023-06-14 at 12 01 32" src="https://github.com/raspberrypi/documentation/assets/53625/52b0765b-a010-43cb-b73a-039dcd9cb082">

But like this on production,

<img width="912" alt="Screenshot 2023-06-14 at 12 01 11" src="https://github.com/raspberrypi/documentation/assets/53625/83e08962-957e-4f44-8d5c-3f73941dae4f">

**Comment by aallan** [comment]
Ping @nelliemckesson 👆

**Comment by aallan** [comment]
Ping also @mudge who might know why it builds locally but not on production.

**Comment by mudge** [comment]
@aallan FWIW it currently looks like this for me on production:

<img width="1250" alt="Screenshot 2023-06-14 at 12 06 48" src="https://github.com/raspberrypi/documentation/assets/287/54fcb301-bde7-4013-86da-f35fd2e86f46">

**Comment by aallan** [comment]
Caching?

**Comment by aallan** [comment]
Thanks @mudge, that makes sense!

**Comment by lurch** [comment]
>we didn't forcibly bust any browser caches by incrementing the ver number. In future, if we modify that URL, it should force everyone's browsers to fetch the latest styles.

@mudge Given the pre-processing that we do when building the production site, would it _perhaps_ make sense to do something like:
```
<link rel="stylesheet" href="{{ site.baseurl }}/css/style.css?ver={{ md5sum.style.css }}">
```

(there's no plumbing in place for this yet, it's just a suggestion)

**Comment by mudge** [comment]
@lurch that'd be ideal (we typically "fingerprint" assets like this across raspberrypi.com).

**Comment by lurch** [comment]
> @lurch that'd be ideal

I've created #2991
